### PR TITLE
Reduce duplication in attachments code

### PR DESCRIPTION
**Pull request by javierm**
## References

* We started improving the attachments in pull request #4586

## Objectives

* Make it easier to migrate to Active Storage

### CODE REVIEW

File: spec/lib/tasks/files_spec.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      expect(File.exists?(image.attachment.path)).to be false
+      expect(File.exists?(document.attachment.path)).to be false
+    end
+
+    it "does not delete recent cached attachments" do
+      image = build(:image)
+      document = build(:document)
+
+      image.attachment.save
+      document.attachment.save
```
[the remark this answers is not in the document]
Comment (javierm) [reply spec/lib/tasks/files_spec.rb:28]: False positive. The `save!` method doesn't exist for paperclip attachments.

File: spec/lib/tasks/files_spec.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+      expect(File.exists?(image.attachment.path)).to be false
+      expect(File.exists?(document.attachment.path)).to be false
+    end
+
+    it "does not delete recent cached attachments" do
+      image = build(:image)
+      document = build(:document)
+
+      image.attachment.save
```
[the remark this answers is not in the document]
Comment (javierm) [reply spec/lib/tasks/files_spec.rb:27]: False positive. The `save!` method doesn't exist for paperclip attachments.

File: spec/lib/tasks/files_spec.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      Rake::Task["files:remove_old_cached_attachments"].reenable
+      Rake.application.invoke_task("files:remove_old_cached_attachments")
+    end
+
+    it "deletes old cached attachments" do
+      image = build(:image)
+      document = build(:document)
+
+      image.attachment.save
+      document.attachment.save
```
[the remark this answers is not in the document]
Comment (javierm) [reply spec/lib/tasks/files_spec.rb:15]: False positive. The `save!` method doesn't exist for paperclip attachments.

File: spec/lib/tasks/files_spec.rb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    let(:run_rake_task) do
+      Rake::Task["files:remove_old_cached_attachments"].reenable
+      Rake.application.invoke_task("files:remove_old_cached_attachments")
+    end
+
+    it "deletes old cached attachments" do
+      image = build(:image)
+      document = build(:document)
+
+      image.attachment.save
```
[the remark this answers is not in the document]
Comment (javierm) [reply spec/lib/tasks/files_spec.rb:14]: False positive. The `save!` method doesn't exist for paperclip attachments.

File: app/components/attachable/fields_component.rb [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+    def file_name
+      attachable.attachment_file_name
+    end
+
+    def destroy_link
+      if !attachable.persisted? && attachable.cached_attachment.present?
+        link_to t("#{plural_name}.form.delete_button"), "#", class: "delete remove-cached-attachment"
+      else
+        link_to_remove_association attachable.new_record? ? t("documents.form.cancel_button") : t("#{plural_name}.form.delete_button"), f, class: "delete remove-#{singular_name}"
```
[the remark this answers is not in the document]
Comment (Senen) [reply app/components/attachable/fields_component.rb:34]: We can extract a method to get the link text:
```suggestion
        link_to_remove_association destroy_attachment_text, f, class: "delete remove-#{singular_name}"
```

and then the new method could be:

```
    def destroy_attachment_text
      attachable.new_record? ? t("documents.form.cancel_button") : t("#{plural_name}.form.delete_button")
    end
```

What do you think @javierm? 🤔

**Review by Senen — COMMENTED** [review]
Hi @javierm, the only thing I found is about [this Rubocop offense](https://github.com/consul/consul/pull/4595/files#r676846176).

**Comment by javierm** [comment]
@sendero I've [applied your suggestion](https://github.com/consul/consul/compare/507fa7c5f5ce..7bbaad9a19). Thanks!

**Review by Senen — APPROVED** [review]

**Pull request merged**